media/js/base/fxa-attribution.es6.js

```diff
@@ -319,38 +382,59 @@ FxaAttribution.appendToProductURL = (url, data) => {
     );
 };
 
-FxaAttribution.getAttributionData = (urlParams) => {
+FxaAttribution.getAttributionData = (urlObject) => {
+    const urlParams = urlObject ? urlObject : {};
     let params = {};
-    const utmData = FxaAttribution.getUtmData(urlParams);
-    const experimentData = FxaAttribution.getExperimentData(urlParams);
-    const couponData = FxaAttribution.getCouponData(urlParams);
-
-    // If there are UTM params in the page URL, then
-    // validate those as referral data.
-    if (utmData) {
-        params = Object.assign(params, utmData);
-    } else {
-        // Check to see if there's a referral cookie.
-        const linkData = FxaAttribution.getFxALinkReferralData();
+    const couponParam = FxaAttribution.getCouponParam(urlParams);
 
-        if (linkData) {
-            params = Object.assign(params, linkData);
+    // Always pass a coupon when present as a parameter.
+    if (couponParam) {
+        FxaAttribution.setCouponStorage(couponParam);
+    }
+
+    // Make sure the visitor hasn't opted out of attribution
+    if (FxaAttribution.isAttributionPermitted()) {
+        const referralData = FxaAttribution.getReferralStorage();
+
+        if (referralData) {
+            params = referralData;
         } else {
-            // Check to for search engine referral data.
-            const searchData = FxaAttribution.getSearchReferralData();
+            const utmData = FxaAttribution.getUtmData(urlParams);
+            const experimentData = FxaAttribution.getExperimentData(urlParams);
+
+            // If there are UTM params in the page URL, then
+            // validate those as referral data.
```

Comment:
So someone with stored UTM info will have that over-ridden if they visit with new UTMs in the URL?

Reply:
If someone has stored referral data, that will always be used (so their first entry page to the site). The data persists from when the active tab was opened to when it gets closed again.

It's possible that someone might enter twice, through two separate campaigns, during that tab's lifetime. In this case, their first entry point would be attributed.

Does that make sense?